addons/Omen/Omen.lua

```diff
@@ -0,0 +1,259 @@
+	local objective = objectives[tonumber(original:match("^%d"))]
+    if mode == 161 then -- Omen messages are 161 color, except total time extension messages which are 121 and irrelevant
+        if string.match(original,"^%d") then
+            for k,v in pairs (messages) do
+                if string.find(original,v.init) then
+                    if objective.mes ~= tonumber(v.id) then -- New Objective
+                        objective.amt = 0
+                    end
+                    objective.mes = tonumber(v.id)
+                    objective.req = tonumber(string.sub(original:match(v.check),1,-2))				
+                elseif string.find(original,v.eval) then
+                    objective.amt = tonumber(string.sub(original:match(v.check),1,-2))
+                    if objective.mes == 0 then -- if loading mid-floor
+                        objective.mes = tonumber(v.id)
+                        objective.req = -1
+                    end
+                end
+                refresh()
+            end
+        elseif string.find(original,"%d+ omen") then
+            omens = original:match("%d+")
+            refresh()
+        elseif string.find(original,"You have %d+ seconds remaining.") then
+            if obj_time == 0 then
+                obj_time = tonumber(original:match("%d+"))
+                end_time = os.time() + obj_time
+                refresh()			
+            end
+        elseif string.find(original,"A spectral light flares up.") then
+            floor_clear = good_col
+            refresh()
+            windower.play_sound(windower.addon_path..'big_clear.wav')
+        elseif string.find(original,"A faint light twinkles into existence.") then
+            windower.play_sound(windower.addon_path..'small_clear.wav')
+        elseif string.find(original,"Vanquish") or string.find(original,"Open %d treasure portent") then
+            local str1 = string.gsub(original,string.char(0x7f).."1","")
+            local str1 = string.gsub(str1,"%p","")			
+            local str1 = string.gsub(str1,"(%s%a)",string.upper)
+            floor_obj = string.gsub(str1,"The","the")
+            if floor_clear == good_col then
+                reset_objectives()
+            end
+            refresh()
+        elseif string.find(original,"The light shall come even if you fail to obey.") then
+            floor_obj = "Free Floor!"
+            if floor_clear == good_col then
+                reset_objectives()
+            end
+            refresh()
+        end		
+    end
+end)
+
+windower.register_event('addon command',function(command)
+    command = command and command:lower() or 'help'
+    if texts.visible(image) then
+        image:hide()
+    else
+        image:show()
+    end
+end)
+
+messages = {
+[1] = {id="1",long="Weapon Skill Damage",short="WS Damage",check="%d+%su",
+init="%d: Reduce your foe's HP by %a*%s*%a*%s*%d+ using a single weapon skill.",
```

Comment:
The bug you mentioned could be caused by incorrect number matching. Is the `%d` here meant to represent the floor number? Because if so, that would be wrong, it should be `%d+`, since `%d` will only match a single digit. So `10` would be matched as `0`. I would try replacing all the `%d` in the following lines with `%d+`.